pkg/aws/metadata/handler_role_name_test.go

```diff
@@ -0,0 +1,88 @@
+package metadata
+
+import (
+	"context"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/uswitch/kiam/pkg/server"
+	st "github.com/uswitch/kiam/pkg/testutil/server"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestReturnRoleWhenClientResponds(t *testing.T) {
+	r, _ := http.NewRequest("GET", "/latest/meta-data/iam/security-credentials/", nil)
```

Comment:
All of these tests use `/latest/meta-data/iam/security-credentials/`. However, when you curl on an ec2 instance the `/latest/meta-data/iam/security-credentials` in at least some situations you don't get `200`:

```
curl -v 169.254.169.254/latest/meta-data/iam/security-credentials
* About to connect() to 169.254.169.254 port 80 (#0)
*   Trying 169.254.169.254...
* Connected to 169.254.169.254 (169.254.169.254) port 80 (#0)
> GET /latest/meta-data/iam/security-credentials HTTP/1.1
> User-Agent: curl/7.29.0
> Host: 169.254.169.254
> Accept: */*
>
* HTTP 1.0, assume close after body
< HTTP/1.0 301 Moved Permanently
< Location: http://169.254.169.254/latest/meta-data/iam/security-credentials/
< Content-Length: 0
< Connection: close
< Date: Mon, 09 Jul 2018 21:12:44 GMT
< Server: EC2ws
```
I don't think the SDK actually uses this endpoint, but should we make this match?

Reply:
I believe you're getting a 301 rather than 200 because you didn't including the trailing slash. See [this Google blog post](https://webmasters.googleblog.com/2010/04/to-slash-or-not-to-slash.html): 
> don’t both return a 200 response code, but that one version redirects to the other.

Reply:
Yeah I think @moofish32 was pointing out that the metadata server they tested on 301'ed, rather than returning credentials. Kiam currently supports either form and serves 200s on both.

@moofish32 I seem to remember others mentioning this stuff varied between instance generations (so it changed between m4 and m5 for example).

Do you remember the instance type you tested this on?

Either way I suspect we should probably add an issue to change it so that it behaves the same way- 301'ing rather than successfully working on both?

Reply:
I've now added #116 to track this issue separately.